src/Magnum/DartIntegration/World.cpp

```diff
@@ -0,0 +1,170 @@
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "World.h"
+
+#include <Magnum/Buffer.h>
+#include <Magnum/Mesh.h>
+#include <Magnum/Texture.h>
+#include <Magnum/Trade/PhongMaterialData.h>
+
+namespace Magnum { namespace DartIntegration {
+World::World(SceneGraph::AbstractBasicObject3D<Float>& object, std::shared_ptr<dart::simulation::World> world): _object(object), _manager{MAGNUM_PLUGINS_IMPORTER_DIR}, _dartWorld(world) {
+    /* load Assimp importer */
+    _importer = _manager.loadAndInstantiate("AssimpImporter");
+}
+
+World::~World() = default;
+
+World& World::refresh() {
+    if (!_dartWorld)
+        return *this;
+    /* clear update flags */
+    for(auto& obj: _dartToMagnum)
+        obj.second->clearUpdateFlag();
+
+    /* parse all skeletons in _dartWorld */
+    for(size_t i = 0; i < _dartWorld->getNumSkeletons(); i++) {
+        parseSkeleton(_object, *_dartWorld->getSkeleton(i));
+    }
+
+    /* remove unused objects */
+    clearUnusedObjects();
+
+    return *this;
+}
+
+World& World::step() {
+    _dartWorld->step();
+    return *this;
+}
+
+World& World::clearUnusedObjects() {
+    std::vector<dart::dynamics::Frame*> unusedFrames;
+
+    /* Find unused objects */
+    for(auto& object_pair: _dartToMagnum)
+    {
+        if(object_pair.second && !object_pair.second->isUpdated())
+            unusedFrames.push_back(object_pair.first);
+    }
+
+    /* Clear unused Objects */
+    _toRemove.clear();
+    for(dart::dynamics::Frame* frame: unusedFrames)
+    {
+        auto it = _dartToMagnum.find(frame);
+        _toRemove.emplace_back(std::move(it->second));
+        _dartToMagnum.erase(it);
+    }
+
+    return *this;
+}
+
+std::vector<std::reference_wrapper<Object>> World::objects() {
+    std::vector<std::reference_wrapper<Object>> objs;
+    for(auto& obj: _dartToMagnum)
+        objs.emplace_back(std::ref(*obj.second));
+
+    return objs;
+}
+
+std::vector<std::reference_wrapper<Object>> World::shapeObjects() {
+    std::vector<std::reference_wrapper<Object>> objs;
+    for(auto& obj: _dartToMagnum)
+        if(obj.second->shapeNode())
+            objs.emplace_back(std::ref(*obj.second));
+
+    return objs;
+}
+
+std::vector<Object*> World::updatedShapeObjects() {
+    std::vector<Object*> objs;
+    objs.insert(objs.end(), _updatedShapeObjects.begin(), _updatedShapeObjects.end());
+    return objs;
+}
+
+World& World::clearUpdatedShapeObjects() {
+    _updatedShapeObjects.clear();
+    return *this;
+}
+
+std::vector<std::reference_wrapper<Object>> World::bodyObjects() {
+    std::vector<std::reference_wrapper<Object>> objs;
+    for(auto& obj: _dartToMagnum)
+        if(obj.second->bodyNode())
+            objs.emplace_back(std::ref(*obj.second));
+
+    return objs;
+}
+
+std::reference_wrapper<Object> World::objectFromDartFrame(dart::dynamics::Frame* frame) {
+    return std::ref(*_dartToMagnum[frame]);
+}
```

Comment:
Sorry to sound like a broken record here: you can return just `Object&` here, no need for the wrapper ;)

Also: since this is a public API at the moment, think about what happens when you ask for a `frame` that's not in the `_dartToMagnum` structure: it will insert a new element with such key, default-construct it (so you get `nullptr` value) and then you dereference it ... :boom: 

Suggestion:

- I would use `at()` instead of `operator[]`, as that does not insert (but rather throw) when not found
- I would explicitly mention the behavior in the docs